feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -0,0 +1,290 @@
+	"github.com/openconfig/featureprofiles/internal/attrs"
+	"github.com/openconfig/featureprofiles/internal/confirm"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/otgutils"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	otgtelemetry "github.com/openconfig/ondatra/gnmi/otg"
+	"github.com/openconfig/ondatra/otg"
+	"github.com/openconfig/ygnmi/ygnmi"
+)
+
+type lldpTestParameters struct {
+	systemName string
+	macAddress string
+	otgName    string
+}
+
+type lldpNeighbors struct {
+	systemName    string
+	portId        string
+	portIdType    otgtelemetry.E_LldpNeighbor_PortIdType
+	chassisId     string
+	chassisIdType otgtelemetry.E_LldpNeighbor_ChassisIdType
+}
+
+const (
+	portName = "port1"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+var (
+	// Ethernet configuration.
+	ateSrc = attrs.Attributes{
+		Name: "ateSrc",
+		MAC:  "02:00:01:01:01:01",
+	}
+
+	// LLDP configuration.
+	lldpSrc = lldpTestParameters{
+		systemName: "ixia-otg",
+		macAddress: "02:00:22:01:01:01",
+		otgName:    "ixia-otg",
+	}
+)
+
+// Determine LLDP advertisement and reception operates correctly.
+func TestCoreLLDPTLVPopulation(t *testing.T) {
+	tests := []struct {
+		desc        string
+		lldpEnabled bool
+	}{
+		{
+			desc:        "LLDP Enabled",
+			lldpEnabled: true,
+		},
+		{
+			desc:        "LLDP Disabled",
+			lldpEnabled: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+
+			// DUT configuration.
+			t.Log("Configure DUT.")
+			dut, dutConf := configureDUT(t, "dut", test.lldpEnabled)
+			dutPort := dut.Port(t, portName)
+			verifyNodeConfig(t, dut, dutPort, dutConf, test.lldpEnabled)
+
+			// ATE Configuration.
+			t.Log("Configure ATE")
+			ate := ondatra.ATE(t, "ate")
+			otg := ate.OTG()
+			otgConfig := configureATE(t, otg)
+
+			checkLLDPMetricsOTG(t, otg, otgConfig, test.lldpEnabled)
+
+			dutPeerState := lldpNeighbors{
+				systemName:    lldpSrc.systemName,
+				chassisId:     lldpSrc.macAddress,
+				chassisIdType: otgtelemetry.LldpNeighbor_ChassisIdType_MAC_ADDRESS,
+				portId:        ate.Port(t, portName).Name(),
+				portIdType:    otgtelemetry.LldpNeighbor_PortIdType_INTERFACE_NAME,
+			}
+
+			if test.lldpEnabled {
+				expOtgLLDPNeighbor := lldpNeighbors{
+					systemName:    dutConf.GetSystemName(),
+					portId:        dutPort.Name(),
+					portIdType:    otgtelemetry.LldpNeighbor_PortIdType_INTERFACE_NAME,
+					chassisId:     strings.ToUpper(dutConf.GetChassisId()),
+					chassisIdType: otgtelemetry.E_LldpNeighbor_ChassisIdType(dutConf.GetChassisIdType()),
+				}
+				checkOTGLLDPNeighbor(t, otg, otgConfig, expOtgLLDPNeighbor)
+				verifyDUTTelemetry(t, dut, dutPort, dutConf, dutPeerState)
+			} else {
+				expOtgLLDPNeighbor := lldpNeighbors{}
+				checkOTGLLDPNeighbor(t, otg, otgConfig, expOtgLLDPNeighbor)
+			}
+		})
+	}
+}
```

Comment:
On second thought, it might even be better to split the TestCoreLLDPTLVPopulation into two, according to the link I shared before. What do you think?

See: https://github.com/golang/go/wiki/TestComments#table-driven-tests-vs-multiple-test-functions

Would you mind to do the same for the ATE test as well?

```suggestion
const (
	lldpEnabled = true
	lldpDisabled = false
)

// TestLLDPEnabled tests LLDP advertisement turned on.
func TestLLDPEnabled(t *testing.T) {
	// DUT configuration.
	t.Log("Configure DUT.")
	dut, dutConf := configureDUT(t, "dut", lldpEnabled)
	dutPort := dut.Port(t, portName)
	verifyNodeConfig(t, dut, dutPort, dutConf, lldpEnabled)

	// ATE Configuration.
	t.Log("Configure ATE.")
	ate := ondatra.ATE(t, "ate")
	otg := ate.OTG()
	otgConfig := configureATE(t, otg)

	checkLLDPMetricsOTG(t, otg, otgConfig, lldpEnabled)

	dutPeerState := lldpNeighbors{
		systemName:    lldpSrc.systemName,
		chassisId:     lldpSrc.macAddress,
		chassisIdType: otgtelemetry.LldpNeighbor_ChassisIdType_MAC_ADDRESS,
		portId:        ate.Port(t, portName).Name(),
		portIdType:    otgtelemetry.LldpNeighbor_PortIdType_INTERFACE_NAME,
	}
	verifyDUTTelemetry(t, dut, dutPort, dutConf, dutPeerState)

	expOtgLLDPNeighbor := lldpNeighbors{
		systemName:    dutConf.GetSystemName(),
		portId:        dutPort.Name(),
		portIdType:    otgtelemetry.LldpNeighbor_PortIdType_INTERFACE_NAME,
		chassisId:     strings.ToUpper(dutConf.GetChassisId()),
		chassisIdType: otgtelemetry.E_LldpNeighbor_ChassisIdType(dutConf.GetChassisIdType()),
	}
	checkOTGLLDPNeighbor(t, otg, otgConfig, expOtgLLDPNeighbor)
}

// TestLLDPDisabled tests LLDP advertisement turned off.
func TestLLDPDisabled(t *testing.T) {
	// DUT configuration.
	t.Log("Configure DUT.")
	dut, dutConf := configureDUT(t, "dut", lldpDisabled)
	dutPort := dut.Port(t, portName)
	verifyNodeConfig(t, dut, dutPort, dutConf, lldpDisabled)

	// ATE Configuration.
	t.Log("Configure ATE.")
	ate := ondatra.ATE(t, "ate")
	otg := ate.OTG()
	otgConfig := configureATE(t, otg)

	checkLLDPMetricsOTG(t, otg, otgConfig, lldpDisabled)
	expOtgLLDPNeighbor := lldpNeighbors{}
	checkOTGLLDPNeighbor(t, otg, otgConfig, expOtgLLDPNeighbor)
}
```

Reply:
Your suggested change was merged. I can't make any changes to the ATE test because I don't have access to a setup where it can be verified.